Integrate PyTorch quantization APIs into ensemble export modules

jamesr66a opened the pull request:
Summary: This gives us a boolean flag `quantize` on the `BeamSearch` module that allows us to apply FBGEMM quantization to a pretrained PyTorch model and export this to PyTorch native runtime.

Differential Revision: D13514776

apaszke commented on torch/csrc/jit/script/module.h:
```diff
@@ -74,6 +74,10 @@ struct Method {
     get_executor().run(stack);
   }
 
+  void run(Stack&& stack) {
```
I don't think we want to add this method. TorchScript has almost no externally visible side effects, so you really only want to run it with a stack that will survive this call, so you can read out its outputs.

Commit:
Fork/join parallelism for ensemble export modules

Differential Revision: D13827861

fbshipit-source-id: 6a227d9ea91be2b7a881e7aadec93eb545b49ef3

Commit:
Add an API to set the number of threads in C10 thread pool (#16669)

Summary:
Tested locally on machine translation service
Pull Request resolved: https://github.com/pytorch/pytorch/pull/16669

Differential Revision: D13927858

fbshipit-source-id: e2573f91fb5cd434da20fa1ac9ba7ca2b6c38e02

Commit:
Integrate PyTorch quantization APIs into ensemble export modules (#309)

Summary:
Pull Request resolved: https://github.com/pytorch/translate/pull/309

Pull Request resolved: https://github.com/pytorch/pytorch/pull/16481

This gives us a boolean flag `quantize` on the `BeamSearch` module that allows us to apply FBGEMM quantization to a pretrained PyTorch model and export this to PyTorch native runtime.

Reviewed By: jmp84

Differential Revision: D13514776

fbshipit-source-id: e250c7b2526dab36d231d42514dbd5eea23e4009
